### ISSUE TITLE
Problem with SDL2 macOS build

### ISSUE THREAD

**Issue by emendelson** [joncampbell123/dosbox-x#2199]
This is slightly complicated, but here goes:

If I build current code for SDL1 under macOS Big Sur (on Intel), the build runs perfectly when I run it in macOS Mojave.

But if I build current code for SDL2 under Big Sur, it runs perfectly under Big Sur, but crashes in Mojave with this error:

```
Dyld Error Message:
  Symbol not found: _objc_opt_class
  Referenced from: /Users/USER/Desktop/dosbox-x (which was built for Mac OS X 11.0)
  Expected in: /usr/lib/libobjc.A.dylib
```

Is there a way to tell the build script to build for earlier versions of macOS? Again, this problem doesn't occur with the SDL1 build.

**Comment by Wengier** [comment]
@emendelson I do not have macOS Big Sur, but maybe this will help?

https://forum.juce.com/t/plugin-fails-to-resolve-symbol-in-libc-in-older-version-of-mac-than-the-build-host/41083

**Comment by emendelson** [comment]
@Wengier Yes - that's exactly what I would hope could be done for the code. But I can't figure out how to modify either the build script or anything else to specify the build-for version. I hope someone here might know the answer.

**Comment by Wengier** [comment]
@emendelson Have you tried to set CMAKE_OSX_DEPLOYMENT_TARGET for the macOS target version? You can find some answers related to this below:

https://stackoverflow.com/questions/34208360/cmake-seems-to-ignore-cmake-osx-deployment-target

https://stackoverflow.com/questions/55254398/why-the-cmake-osx-deployment-target-not-work-when-using-cmake-generate-xcode-pro

**Comment by emendelson** [comment]
@Wengier I added this line to the top of every cmakelists.txt file in the source code:

`set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING`

The same problem still occurred.

**Comment by emendelson** [comment]
I added this line to the top of every cmakelists.txt file in the code:

`set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING "Minimum OS X deployment version")`

The same crash still occurred in Mojave.

**Comment by Wengier** [comment]
@emendelson Have you tried adding FORCE in the end as mentioned in a link above?

**Comment by emendelson** [comment]
@Wengier That didn't work either. I still get this error:

```
Dyld Error Message:
  Symbol not found: _objc_opt_class
  Referenced from: /Users/USER/Desktop/dosbox-x (which was built for Mac OS X 11.0)
  Expected in: /usr/lib/libobjc.A.dylib
```

**Comment by Wengier** [comment]
Hi i

**Comment by Wengier** [comment]
@emendelson Since I am not really familiar with macOS, you may want to ask it in the StackOverFlow discussion, which may be helpful. Apparently this issue is not specific to DOSBox-X.

**Comment by emendelson** [comment]
@Wengier - Have asked, but no reply yet.

**Comment by joncampbell123** [comment]
I'm able to compile on Mac OS X just fine. Can you try just using git clean -dfx and git reset to clean up your copy, and then run build-macosx again?

**Comment by emendelson** [comment]
@joncampbell123 - It compiles just fine. The problem is this: if I compile SDL2 on Big Sur and run the resulting executable under Mojave it crashes when I type anything at the DOS prompt. This makes it impossible to compile an SDL2 version on Big Sur that can be used by anyone with an older version.

This does not happen if I compile SDL1

**Comment by joncampbell123** [comment]
@emendelson Bizarre, I thought Mac OS X would be more forward compatible than that. Maybe it's an issue with SDL2?

As I don't have those older versions of OS X around anymore, I can't debug that myself.

If you run SDL2 builds from the OS X equivalent of gdb and let it crash that way, does the backtrace give any clues?

**Comment by emendelson** [comment]
> As I don't have those older versions of OS X around anymore, I can't debug that myself.
> 
> If you run SDL2 builds from the OS X equivalent of gdb and let it crash that way, does the backtrace give any clues?

@joncampbell123 - I have older versions running in Parallels Desktop and VMware Fusion, not on hardware.

I'm afraid I don't know how to run the OS X equivalent of gdb (or any version of gdb), but I can try to teach myself and will report.
